javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java

```diff
@@ -671,8 +671,10 @@ private void doExtract(FileExtractor extractor, Path file, ExtractorState state)
 			long start = logBeginProcess("Extracting " + file);
 			extractor.extract(f, state);
 			logEndProcess(start, "Done extracting " + file);
-		} catch (IOException e) {
-			throw new ResourceError("Exception while extracting " + file + ".", e);
+		} catch (Throwable t) {
+			System.err.println("Exception while extracting " + file + ".");
+			t.printStackTrace(System.err);
+			System.exit(1);
```

Comment:
Subjective: It seems dangerous to have a `System.exit(1)` call so many calls away from `main(string[] args)`, or other potential callers of `new AutoBuild().run()` that may not expect a full VM shutdown.
Be aware that unless `System.exit(1)` throws an exception, it will bypass enclosing `catch` and `finally` blocks.

Reply:
That is a fair point. I originally introduced this because I was experiencing problems with exceptions thrown by worker threads being silently swallowed, which made problems hard to diagnose. I believe this should actually not be necessary anymore due to https://github.com/Semmle/ql/pull/794/commits/7765c6d486eec5fe58abc1f09140e88bcaa9439a, so if the distribution upgrade doesn't show any issues I'll revert this change.